src/parser/jobs/smn/modules/Summons.tsx

```diff
@@ -313,31 +313,57 @@ export class Summons extends Analyser {
 				key: summon.start,
 				title: {
 					content: <>
-						{this.parser.formatEpochTimestamp(summon.start)}: <ActionLink {...this.data.getAction(summon.data.demiSummon.action)} />
+						{<span style={{ width: '200px', display: 'inline-block' }}>{this.parser.formatEpochTimestamp(summon.start)}: <ActionLink {...this.data.getAction(summon.data.demiSummon.action)} /></span>}
+						{this.getEgiIcons(summon)}
 					</>,
 				},
-				content: {content: data.display},
+				content: { content: data.display },
 			},
 			hasError: data.hasError,
 		}
 	}
 
-	private buildWindowOutput(summon: HistoryEntry<SummonWindow>) {
-		const demiRow = (summon.data.demiSummon?.action === this.data.actions.SUMMON_BAHAMUT.id) ?
-			this.buildBahamutRow(summon) :
-			this.buildPhoenixRow(summon)
+	private getEgiIcons(summon: HistoryEntry<SummonWindow>) {
+		const egis = [summon.data.titanSummon, summon.data.ifritSummon, summon.data.garudaSummon]
+		egis.sort((a, b) => {
+			if (a === undefined) { return -1 }
+			if (b === undefined) { return 1 }
+			return (a.timestamp > b.timestamp) ? 1 : -1
+		})
+
+		const retval: JSX.Element[] = []
+		egis.forEach(egi => {
+			if (egi !== undefined) {
+				if (egi === summon.data.ifritSummon) {
+					retval.push(< > <ActionLink showName={false} action="SUMMON_IFRIT_II" />&nbsp;{this.printUsageTime(summon.data.ifritSummon)} </>)
```

Comment:
I think putting the egi timestamps here is not a net positive.  Instead we should just show the icons.

It's subtle, but with the timestamps here, I have to scan the row more to see the egi order, whereas with just the single space, I can see them all at once.  Similarly, I assume that when someone is looking at these, the important bit is the order and the precise timing is rarely what they are looking for.

Reply:
Should I put egi -> egi -> egi or just egi egi egi

Reply:
I'd lean towards egi egi egi.  It seems to give enough space between them and minimizes clutter and space needed.